Sources/Common/Background Queue/Queue.swift

```diff
@@ -48,42 +49,29 @@ public protocol Queue: AutoMockable {
         // sourcery:TypeCast="AnyEncodable(data)"
         data: TaskData,
         groupStart: QueueTaskGroup?,
-        blockingGroups: [QueueTaskGroup]?
-    ) -> ModifyQueueResult
+        blockingGroups: [QueueTaskGroup]?,
+        onComplete: @escaping (ModifyQueueResult) -> Void
+    )
 
     func run(onComplete: @escaping () -> Void)
 
     func deleteExpiredTasks()
 }
 
 public extension Queue {
-    func addTask<TaskData: Codable>(
-        type: String,
-        // sourcery:Type=AnyEncodable
-        // sourcery:TypeCast="AnyEncodable(data)"
-        data: TaskData
-    ) -> ModifyQueueResult {
-        addTask(type: type, data: data, groupStart: nil, blockingGroups: nil)
-    }
-
-    func addTask<TaskData: Codable>(
-        type: String,
-        // sourcery:Type=AnyEncodable
-        // sourcery:TypeCast="AnyEncodable(data)"
-        data: TaskData,
-        groupStart: QueueTaskGroup?
-    ) -> ModifyQueueResult {
-        addTask(type: type, data: data, groupStart: groupStart, blockingGroups: nil)
-    }
-
+    // convenient addTask alternative that doesn't require you give a value for all parameters
     func addTask<TaskData: Codable>(
         type: String,
         // sourcery:Type=AnyEncodable
         // sourcery:TypeCast="AnyEncodable(data)"
         data: TaskData,
-        blockingGroups: [QueueTaskGroup]?
-    ) -> ModifyQueueResult {
-        addTask(type: type, data: data, groupStart: nil, blockingGroups: blockingGroups)
+        groupStart: QueueTaskGroup? = nil,
+        blockingGroups: [QueueTaskGroup]? = nil,
+        onComplete: ((ModifyQueueResult) -> Void)? = nil
```

Comment:
I don't understand what value that would provide?

Reply:
Comes down to coding preference I believe, I find non-nullable properties easier to handle, maintain, and more consistent. You don't have to add unnecessarily `?` next to variables. And also would provide some consistency with the Android(Kotlin) counterpart which prefers to avoid `null`. 

But like i mentioned earlier, its a style preference not a blocker for PR.